File: examples/scale/scale.js
function setup(c) {
  //c.noLoop();
  c.cellPadding(" ");
}

function draw(c) {
  c.clear();
  c.stroke(".", "#ffffff", "#ffffff");
  c.fill(".", "#ffffff", "#000000");
  c.translate(c.width / 2, c.height / 2);
  c.scale(c.map(Math.sin(c.framecount * 0.1), -1, 1, 1, 4));
  c.rotate(c.framecount * 0.008);
  c.circle(0, 0, 5);
  c.circle(10, 10, 5);
}

File: examples/scale/scale.go
package main

import (
	"context"
	"math"

	"github.com/emprcl/runal"
)

func main() {
	runal.Run(context.Background(), setup, draw)
}

func setup(c *runal.Canvas) {
	//c.NoLoop();
	c.CellPadding(" ")
}

func draw(c *runal.Canvas) {
	c.Clear()
	c.Stroke(".", "#ffffff", "#ffffff")
	c.Fill(".", "#ffffff", "#000000")
	c.Translate(c.Width/2, c.Height/2)
	c.Scale(c.Map(math.Sin(float64(c.Framecount)*0.1), -1, 1, 1, 4))
	c.Rotate(float64(c.Framecount) * 0.008)
	c.Circle(0, 0, 5)
	c.Circle(10, 10, 5)
}
